Importing Libraries

In [3]:
import pandas as pd
import numpy as np

Loading the Data set.

In [6]:
df = pd.read_csv("Dataset.csv")
df

Unnamed: 0  AirFlowRate  DS18B20sen1_temperatureC  \
0              0      0.56249                    9.5625   
1              1      1.02559                    9.5625   
2              2      1.46180                    9.5625   
3              3      1.34875                    9.5625   
4              4      1.54852                    9.5625   
...          ...          ...                       ...   
3574        3574      0.00000                   22.0625   
3575        3575      0.00000                   22.5625   
3576        3576      0.00000                   23.0000   
3577        3577      0.06978                   23.3125   
3578        3578      0.10362                   23.6250   

      DS18B20sen2_temperatureC  ct_curr    ct_power    ct_volt  \
0                      17.0625    7.632  1717.69995  229.30000   
1                      17.1250    7.597  1708.90002  229.30000   
2                      17.0625    7.683  1730.09998  229.50000   
3                      17.1250    7.628  1717.90002  229.50000   
4                      17.0625    7.607  1711.00000  229.39999   
...                        ...      ...         ...        ...   
3574                   23.3750    0.000     0.00000  232.80000   
3575                   23.6875    0.000     0.00000  232.89999   
3576                   24.1250    0.000     0.00000  233.00000   
3577                   24.3125    0.000     0.00000  233.00000   
3578                   24.5000    0.000     0.00000  233.10001   

                        time                timestamp  BME280_HUMIDI  \
0            0 days 00:00:00  2024-05-15 13:30:00.000      41.872580   
1     0 days 00:00:46.750000  2024-05-15 13:30:46.750      42.127048   
2     0 days 00:01:33.500000  2024-05-15 13:31:33.500      42.249997   
3     0 days 00:02:20.250000  2024-05-15 13:32:20.250      42.248148   
4            0 days 00:03:07  2024-05-15 13:33:07.000      42.366430   
...                      ...                      ...            ...   
3574  0 days 01:33:06.704000  2024-05-31 17:48:06.704      55.243160   
3575  0 days 01:33:25.578000  2024-05-31 17:48:25.578      56.885740   
3576  0 days 01:33:44.452000  2024-05-31 17:48:44.452      57.664060   
3577  0 days 01:34:03.326000  2024-05-31 17:49:03.326      57.103520   
3578  0 days 01:34:22.200000  2024-05-31 17:49:22.200      57.378910   

      BME280_TEMP  temperature  
0       43.204799         16.0  
1       43.079669         16.0  
2       43.024747         16.0  
3       43.027283         16.0  
4       42.965117         16.0  
...           ...          ...  
3574    37.600000         26.0  
3575    36.930000         26.0  
3576    36.770000         26.0  
3577    37.010000         26.0  
3578    36.800000         26.0  

[3579 rows x 12 columns]

Consists of 12 columns with 3579 datarows.

Understanding the dataset.

In [7]:
df.head()

Unnamed: 0  AirFlowRate  DS18B20sen1_temperatureC  \
0           0      0.56249                    9.5625   
1           1      1.02559                    9.5625   
2           2      1.46180                    9.5625   
3           3      1.34875                    9.5625   
4           4      1.54852                    9.5625   

   DS18B20sen2_temperatureC  ct_curr    ct_power    ct_volt  \
0                   17.0625    7.632  1717.69995  229.30000   
1                   17.1250    7.597  1708.90002  229.30000   
2                   17.0625    7.683  1730.09998  229.50000   
3                   17.1250    7.628  1717.90002  229.50000   
4                   17.0625    7.607  1711.00000  229.39999   

                     time                timestamp  BME280_HUMIDI  \
0         0 days 00:00:00  2024-05-15 13:30:00.000      41.872580   
1  0 days 00:00:46.750000  2024-05-15 13:30:46.750      42.127048   
2  0 days 00:01:33.500000  2024-05-15 13:31:33.500      42.249997   
3  0 days 00:02:20.250000  2024-05-15 13:32:20.250      42.248148   
4         0 days 00:03:07  2024-05-15 13:33:07.000      42.366430   

   BME280_TEMP  temperature  
0    43.204799         16.0  
1    43.079669         16.0  
2    43.024747         16.0  
3    43.027283         16.0  
4    42.965117         16.0

In [8]:
df.tail()

Unnamed: 0  AirFlowRate  DS18B20sen1_temperatureC  \
3574        3574      0.00000                   22.0625   
3575        3575      0.00000                   22.5625   
3576        3576      0.00000                   23.0000   
3577        3577      0.06978                   23.3125   
3578        3578      0.10362                   23.6250   

      DS18B20sen2_temperatureC  ct_curr  ct_power    ct_volt  \
3574                   23.3750      0.0       0.0  232.80000   
3575                   23.6875      0.0       0.0  232.89999   
3576                   24.1250      0.0       0.0  233.00000   
3577                   24.3125      0.0       0.0  233.00000   
3578                   24.5000      0.0       0.0  233.10001   

                        time                timestamp  BME280_HUMIDI  \
3574  0 days 01:33:06.704000  2024-05-31 17:48:06.704       55.24316   
3575  0 days 01:33:25.578000  2024-05-31 17:48:25.578       56.88574   
3576  0 days 01:33:44.452000  2024-05-31 17:48:44.452       57.66406   
3577  0 days 01:34:03.326000  2024-05-31 17:49:03.326       57.10352   
3578  0 days 01:34:22.200000  2024-05-31 17:49:22.200       57.37891   

      BME280_TEMP  temperature  
3574        37.60         26.0  
3575        36.93         26.0  
3576        36.77         26.0  
3577        37.01         26.0  
3578        36.80         26.0

In [9]:
df.rename(columns={'DS18B20sen1_temperatureC' : 'Vapor Temperature', 'DS18B20sen2_temperatureC' : 'Liquid Temperature', 'ct_curr' : 'Current','ct_power' : 'Power', 'ct_volt': 'Voltage', 'BME280_HUMIDI': 'Outdoor Humidity', 'BME280_TEMP': 'Outdoor Temperature', 'temperature': 'Indoor Temperature'}, inplace=True)
df.head()

Unnamed: 0  AirFlowRate  Vapor Temperature  Liquid Temperature  Current  \
0           0      0.56249             9.5625             17.0625    7.632   
1           1      1.02559             9.5625             17.1250    7.597   
2           2      1.46180             9.5625             17.0625    7.683   
3           3      1.34875             9.5625             17.1250    7.628   
4           4      1.54852             9.5625             17.0625    7.607   

        Power    Voltage                    time                timestamp  \
0  1717.69995  229.30000         0 days 00:00:00  2024-05-15 13:30:00.000   
1  1708.90002  229.30000  0 days 00:00:46.750000  2024-05-15 13:30:46.750   
2  1730.09998  229.50000  0 days 00:01:33.500000  2024-05-15 13:31:33.500   
3  1717.90002  229.50000  0 days 00:02:20.250000  2024-05-15 13:32:20.250   
4  1711.00000  229.39999         0 days 00:03:07  2024-05-15 13:33:07.000   

   Outdoor Humidity  Outdoor Temperature  Indoor Temperature  
0         41.872580            43.204799                16.0  
1         42.127048            43.079669                16.0  
2         42.249997            43.024747                16.0  
3         42.248148            43.027283                16.0  
4         42.366430            42.965117                16.0